### ISSUE TITLE
it's impossible to throw an error from server with custom error code

### ISSUE THREAD

**Issue by Imperat** [shogowada/json-rpc-2.0#19]
Hello!
Currently if a server method throws an exception this library will respond with object:
```
error: {
  code: 0,
  message: 'Some Error' // This is taken from error object
}
```
I would like to customize error code instead of constantly having it equaled to zero to let clients distinguish different kinds of errors: authorization errors, internal server errors and etc.

**Comment by Imperat** [comment]
What if we change this fragment:

```
const mapErrorToJSONRPCResponse = (
  id: JSONRPCID | undefined,
  error: any
): JSONRPCResponse | null => {
  if (id !== undefined) {
    return createJSONRPCErrorResponse(
      id,
      DefaultErrorCode,
      (error && error.message) || "An unexpected error occurred"
    );
  } else {
    return null;
  }
};
```

to this one:

```
const mapErrorToJSONRPCResponse = (
  id: JSONRPCID | undefined,
  error: any
): JSONRPCResponse | null => {
  if (id !== undefined) {
    return createJSONRPCErrorResponse(
      id,
      (error && error.code) || DefaultErrorCode,
      (error && error.message) || "An unexpected error occurred"
    );
  } else {
    return null;
  }
};
```
?

**Comment by shogowada** [comment]
Thanks for the suggestion! I actually gave it a thought in the past, but ended up not doing it because of the reason noted at  https://github.com/shogowada/json-rpc-2.0/pull/15#issuecomment-751519798

I wouldn't want to sacrifice discoverability for convenience because in my experience, it actually makes code harder to reason about. I hope this makes sense.

The above link also documents the workaround. Does it suffice your use case?

**Comment by shogowada** [comment]
Or, maybe it can let user ingest it's own `mapErrorToJSONRPCResponse` method?

```typescript
const server = new JSONRPCServer();
server.mapErrorToJSONRPCResponse = (id: JSONRPCID, error: any): JSONRPCErrorResponse => {
  // implement custom mapErrorToJSONRPCResponse
};
```

**Comment by shogowada** [comment]
This change is released under v0.2.17: https://github.com/shogowada/json-rpc-2.0/pull/21

I hope this works for your use case!
